<launch>
  <!-- Enable rviz visualization -->
  <arg name="rviz"        default="true" />
  <!-- Set nodes on debug mode -->
  <arg name="debug"       default="false" />
  <!-- mpc_type 1 for SQP and 0 for DDP  -->
  <arg name="sqp_mpc"     default="1"/>
  <arg name="dummy_sim"   default="0"/>
  <arg name="dummy_sim_arm" default="1"/>
  <arg name="visualize_humanoid" default="true"/>
  <arg name="visualize_mm" default="false"/>

  <!-- The URDF model of the robot -->
  <!-- <arg name="urdfFile"    value="$(find kuavo_assets)/models/biped_s45/urdf/biped_s45_l.urdf" /> -->
  <arg name="urdfFile"    value="$(find kuavo_assets)/models/biped_s45/urdf/biped_s45.urdf" />
  <!-- The task file for the mpc. -->
  <arg name="taskFile"    value="$(find ocs2_mobile_manipulator)/config/kuavo/task.info" />
  <!-- The library folder to generate CppAD codegen into -->
  <arg name="libFolder"   value="$(find ocs2_mobile_manipulator)/auto_generated/kuavo" />

  <!-- make the files into global parameters -->
  <param name="mm/taskFile"   value="$(arg taskFile)" />
  <param name="mm/urdfFile"   value="$(arg urdfFile)" />
  <param name="mm/libFolder"  value="$(arg libFolder)" />
  <param name="mm/mpcType"    value="$(arg sqp_mpc)" />
  <param name="dummy_sim"     value="$(arg dummy_sim)" />
  <param name="dummy_sim_arm" value="$(arg dummy_sim_arm)" />
  <param name="visualize_humanoid" value="$(arg visualize_humanoid)" />
  <param name="visualize_mm"  value="$(arg visualize_mm)" />

  <!-- <group if="$(arg rviz)"> -->
  <include file="$(find ocs2_mobile_manipulator_ros)/launch/include/visualize.launch">
        <arg name="urdfFile" value="$(arg urdfFile)" />
        <arg name="rviz"     value="$(arg rviz)" />
  </include>
  <!-- </group> -->

  <node pkg="mobile_manipulator_controllers" type="mobile_manipulator_controllers_node" name="mobile_manipulator_controllers_node" output="screen" />
  <node pkg="mobile_manipulator_controllers" type="mobile_manipulator_controllers_ik_target" name="mobile_manipulator_controllers_ik_target" output="screen" />

  <node if="$(arg rviz)" pkg="ocs2_mobile_manipulator_ros" type="mobile_manipulator_target" name="mobile_manipulator_target" 
        output="screen" launch-prefix="" />
  <!-- static TO-DO: 高度需要根据rosparam更改(12/13 by matthew) -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="static_tf" args="0 0 0.82 0 0 0 odom world 100" /> -->
  <node pkg="mobile_manipulator_controllers" type="error_monitor.py" name="error_monitor" output="screen" />
</launch>
